### PR TITLE
Fixes for compilation issues

### PR DESCRIPTION
**Pull request by KulikovNikita**

### CODE REVIEW

**Comment by rlnx** [comment]
I'd recommend to not create separate Bazel module for metrics but make it part of test engine

File: cpp/oneapi/dal/algo/knn/BUILD [hunk 1]
```diff
@@ -22,7 +22,7 @@ dal_test_suite(
         "test/*.cpp",
     ]),
     dal_deps = [
-        "@onedal//cpp/oneapi/dal/test/engine/metrics",
+        "@onedal//cpp/oneapi/dal/test/engine/metrics:metrics",
```
Comment (rlnx) [inline cpp/oneapi/dal/algo/knn/BUILD:25]: The change is equivalent to the old line

Comment (rlnx) [reply cpp/oneapi/dal/algo/knn/BUILD:25]: To fix the build you'll need to move deps on test engine to `dal_test_deps`, but as I suggested make metrics part of common test engine module

File: cpp/oneapi/dal/test/engine/metrics/classification.hpp [hunk 2]
```diff
@@ -16,22 +16,18 @@
...
 
-namespace de = oneapi::dal::detail;
-
 namespace oneapi::dal::test::engine {
 
 template <typename Float = double>
-auto accuracy_score(const table& groundtruth,
-                    const table& prediction,
-                    const Float tolerance = Float(0)) {
+inline auto accuracy_score(const table& groundtruth,
```
Comment (mchernov-intel) [inline cpp/oneapi/dal/test/engine/metrics/classification.hpp:28]: Why did you add `inline`?

**Review by mchernov-intel — APPROVED** [review]

**Review by rlnx — APPROVED** [review]

File: cpp/oneapi/dal/test/engine/metrics/classification.hpp (hunk 2 above)
Comment (KulikovNikita) [reply cpp/oneapi/dal/test/engine/metrics/classification.hpp:28]: It's common practice to mark functions as inline in headers to prevent symbol's duplication

**Comment by KulikovNikita** [comment]
http://intel-ci.intel.com/eb7a75dd-ee34-f147-bdbb-8cdcd4b723a1 - Bazel

**Review by napetrov — APPROVED** [review]
